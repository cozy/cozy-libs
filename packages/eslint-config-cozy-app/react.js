'use strict'

const basics = require('./basics')

module.exports = {
  plugins: basics.plugins.concat(['react-hooks']),
  extends: basics.extends.concat(['plugin:react/recommended']),
  parser: basics.parser,
  parserOptions: { ecmaFeatures: { jsx: true } },
  env: basics.env,
  settings: { react: { version: 'latest' } },
  rules: Object.assign({}, basics.rules, {
    'react/prop-types': 'off',
    'react-hooks/rules-of-hooks': 'error',
    'react-hooks/exhaustive-deps': 'warn'
  })
}
