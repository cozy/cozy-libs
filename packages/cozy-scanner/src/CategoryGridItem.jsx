import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import palette from 'cozy-ui/transpiled/react/palette'
import { Icon, IconStack } from 'cozy-ui/transpiled/react'

import styles from './stylesheet.css'

const CategoryGridItem = ({ isSelected, icon, theme, label }) => {
  return (
    <div
      className={classNames(
        styles['grid-item'],
        'u-pb-half u-ph-half u-bxz  u-ellipsis u-c-pointer',
        {
          [styles['border-selected']]: isSelected,
          [styles['grid-item__selected']]: isSelected && label,
          [styles['grid-item__selected__without_label']]: isSelected && !label,
          [styles['border-not-selected']]: !isSelected
        }
      )}
    >
      <IconStack
        foreground={
          icon ? (
            <Icon
              icon={icon}
              color={isSelected ? palette.dodgerBlue : palette.coolGrey}
              size={'16'}
            />
          ) : null
        }
        background={
          <Icon
            icon={'file-duotone'}
            size={'32'}
            color={isSelected ? palette.dodgerBlue : palette.coolGrey}
          />
        }
      />

      <div className="u-flex u-flex-column u-ellipsis u-bxz">
        <span className={classNames(styles['grid-item-theme'], 'u-ellipsis')}>
          {theme}
        </span>
        {label && (
          <span className={classNames(styles['grid-item-label'], 'u-ellipsis')}>
            {label}
          </span>
        )}
      </div>
    </div>
  )
}
CategoryGridItem.propTypes = {
  isSelected: PropTypes.bool.isRequired,
  icon: PropTypes.string,
  theme: PropTypes.string.isRequired,
  label: PropTypes.string
}

export default CategoryGridItem
