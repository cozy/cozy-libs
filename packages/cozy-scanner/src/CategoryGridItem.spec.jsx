import React from 'react'
import { render } from '@testing-library/react'

import CategoryGridItem from './CategoryGridItem'

describe('CategoryGridItem', () => {
  it('should match snapshot if selected and icon', () => {
    const { asFragment } = render(
      <CategoryGridItem
        isSelected={true}
        label={'test'}
        theme={'toto'}
        icon={'car'}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
