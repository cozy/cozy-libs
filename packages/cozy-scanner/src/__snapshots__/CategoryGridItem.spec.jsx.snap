// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryGridItem should match snapshot if selected and icon 1`] = `
<DocumentFragment>
  <div
    class="grid-item u-pb-half u-ph-half u-bxz  u-ellipsis u-c-pointer border-selected grid-item__selected"
  >
    <div
      class="styles__IconStack-wrapper___10dhG"
    >
      <svg
        class="styles__icon___23x3R"
        height="32"
        style="fill: var(--dodgerBlue);"
        width="32"
      >
        <use
          xlink:href="#file-duotone"
        />
      </svg>
      <div
        class="styles__IconStack-foregroundIcon___ZvY-t"
      >
        <svg
          class="styles__icon___23x3R"
          height="16"
          style="fill: var(--dodgerBlue);"
          width="16"
        >
          <use
            xlink:href="#car"
          />
        </svg>
      </div>
    </div>
    <div
      class="u-flex u-flex-column u-ellipsis u-bxz"
    >
      <span
        class="grid-item-theme u-ellipsis"
      >
        toto
      </span>
      <span
        class="grid-item-label u-ellipsis"
      >
        test
      </span>
    </div>
  </div>
</DocumentFragment>
`;
